PI-System-Deployment-Tests/source/PreliminaryChecks/PreliminaryChecks.cs

```diff
@@ -423,6 +424,140 @@ public void CheckManualLoggerHomePageTest()
             }
         }
 
+        /// <summary>
+        /// Checks if the current user has db_datareader, db_datawriter, and execute permissions 
+        /// on PI Manual Logger stored procedures in the PimlWindows database.
+        /// </summary>
+        /// <remarks>
+        /// <para>Test Steps:</para>
+        /// <para>Get SQL Roles assigned to current user</para>
+        /// <para>Check if the db_datareader and db_datawriter roles are present</para>
+        /// <para>Get SQL Objects assigned to the current user</para>
+        /// <para>Check if the SQL Objects have execute permission</para>
+        /// </remarks>
+        [Fact]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Security", "CA2100:Review SQL queries for security vulnerabilities", Justification = "This does not accept any user input so the risk for SQL injection is low.")]
+        public void CheckMinimumManualLoggerSecurity()
+        {
+            var impersonationUserSetting = Settings.PIManualLoggerWebImpersonationUser;
+
+            if (string.IsNullOrEmpty(Settings.PIManualLogger))
+            {
+                Output.WriteLine($"'PIManualLogger' setting value is not set in app.config. Check if minimum security check was skipped.");
+            } 
+            else if (string.IsNullOrEmpty(Settings.PIManualLoggerSQL))
+            {
+                Output.WriteLine($"'PIManualLoggerSQL' setting value is not set in app.config. Check if minimum security check was skipped.");
+            }
+            else
+            {
+                var hasDataReader = false;
+                var hasDataWriter = false;
+                string currentUser;
+
+                if (!string.IsNullOrEmpty(impersonationUserSetting))
+                {
+                    currentUser = impersonationUserSetting;
+                } 
+                else
+                {
+                    currentUser = $@"{Environment.UserDomainName}\{Environment.UserName}";
```

Comment:
Theoretically, a windows user could be constructed with a name that performs a sql injection attack. Allowed characters likely prevent this, but couldn't we just convert this SQL query to use parameters to be safe and use best practices?